app/services/bgs/dependent_service.rb

```diff
@@ -78,5 +97,26 @@ def get_form_hash_686c(file_number:)
         }
       }
     end
+
+    def validate_file_number_format!(file_number:)
+      # We've observed cases where BGS's file numbers are ten or thirteen digits.
+      # BGS has indicated that these file numbers are incorrect and that file numbers
+      # should be eight or nine digits. I want to log cases like this, so we can
+      # tell BGS the kinds of file numbers we are seeing.
+      if file_number.length < 8 || file_number.length > 9
+        file_number_pattern = file_number.gsub(/[0-9]/, 'X')
+        raise "Aborting Form 686c submission: BGS file_nbr has invalid format! (#{file_number_pattern})"
+      end
+    end
+
+    def validate_file_number_matches_ssn!(file_number:)
+      # BGS has indicated that nine-digit file numbers should be equal to the
+      # veteran's SSN. I want to log instances where that is not the case, so that
+      # we can inform MPI and others of instances where veteran account data is
+      # screwed up.
+      if file_number.length == 9 && file_number != ssn
+        raise 'Aborting Form 686c submission: VA.gov SSN does not match BGS file_nbr!'
```

Comment:
It may be useful in this log message to add the `user_uuid` or other info into this message for the log. This is not PII and allows us to tie back to a va.gov user to investigate. 

Could add this to the `validate_file_number_format` log message too so you dont log PII, but can still catch occurances of it, and then use the UUID to get to the (potential) PII/user info, in the DB

Reply:
Ah yeah, good catch. I'll add that in now.

Reply:
actually 🤔 sentry/controller logging may put that in for you... not 100% sure

Reply:
Yeah, I thought that too. But I think may only be on the controller level. I'll add it in just to be safe